### PR TITLE
EVM phase III: events index

### PR DESCRIPTION
**Pull request by droserasprout**

### CODE REVIEW

File: src/dipdup/indexes/evm_subsquid_events/matcher.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    matched_event: SubsquidEventData,
+) -> SubsquidEvent[Any]:
+    """Prepare handler arguments, parse key and value. Schedule callback in executor."""
+    _logger.info('%s: `%s` handler matched!', matched_event.level, handler_config.callback)
+
+    typename = handler_config.contract.module_name
+    event_abi = package.get_evm_events(typename)[handler_config.name]
+    topic1 = decode_hex(matched_event.topics[1] or '')
+    topic2 = decode_hex(matched_event.topics[2] or '')
+
```
Comment (m-kus) [inline src/dipdup/indexes/evm_subsquid_events/matcher.py:35]: + topic3

File: src/dipdup/indexes/evm_subsquid_events/index.py [hunk 2]
```diff
@@ -1,13 +1,109 @@
...
+    def _create_fetcher(self, first_level: int, last_level: int) -> EventLogFetcher:
+        """Get addresses to fetch events during initial synchronization"""
+        addresses = set()
+        topics = set()
+
+        for handler_config in self._config.handlers:
+            addresses.add(handler_config.contract.address)
+
+            event_abi = self._ctx.package.get_evm_events(handler_config.contract.module_name)[handler_config.name]
+            topics.add(event_abi['topic0'])
```
Comment (m-kus) [inline src/dipdup/indexes/evm_subsquid_events/index.py:53]: Maybe make it typed? To do `event_abi.topic0`

File: src/dipdup/models/evm_subsquid.py [hunk 3]
```diff
@@ -1,13 +1,54 @@
...
+class SubsquidEventData:
+    address: str
+    # block_hash: str
+    # block_number: int
+    data: str
+    # index: int
+    # removed: bool
+    topics: tuple[str, ...]
+    # transaction_hash: str
+    # transaction_index: int
```
Comment (m-kus) [inline src/dipdup/models/evm_subsquid.py:33]: Either hash or id required for merging logs into txs (for operation index) — need to check what index is

File: src/dipdup/models/evm_subsquid.py [hunk 4]
```diff
@@ -1,13 +1,54 @@
...
+    # FIXME: temporary field for HasLevel protocol, set somewhere
```
Comment (m-kus) [inline src/dipdup/models/evm_subsquid.py:34]: In GQL there are aliases (need to check if archives support the syntax)
block_number -> level, right?

Comment (droserasprout) [reply src/dipdup/models/evm_subsquid.py]: Property works fine now; I don't think it worth it.

**Review by m-kus — APPROVED** [review]

**Pull request merged**